app/src/androidTest/assets/features/publicPrivateKeyGeneration.feature

```diff
@@ -0,0 +1,15 @@
+Feature: Public and private key generation
+  As a user
+  I want to have the app generate a public/private key pair when I use it for the first time
+  So that I don't have to generate a pair myself
```

Comment:
The idea is that the confirmation is done through user input (think something like the Challenge-Response model). Encryption is not needed here for three reasons:
- The connection itself is already secure. Bluetooth has build-in encryption, NFC is very short range so interception is nearly impossible (other mediums are, admittedly, possibly less secure).
- What the confirmation is trying to accomplish is a guarantee that I actually received your key and not someone else's key.
- The public keys are public after all, it is not that bad if they get out in the open world... All I really care about is that I really have your key and I can really send secure message to you specifically.